### DISCUSSION TITLE
plot initial configuration of Laplace equation

### DISCUSSION THREAD

**Discussion by goldendijon** [zwicker-group/py-pde#607] [Q&A]
Hi,
First, thank you very much for the great package!

I want to solve Laplace equation for the following configuration
![image](https://github.com/user-attachments/assets/e2c47d0e-f4e3-47de-913d-0ec712ab5090)
the boundary conditions are cone at the button and a regular value square value potential at the top.
I wanted to solve this using the CylindricalSymGrid and then the great should be seems like:
![image](https://github.com/user-attachments/assets/c2a7da22-f828-4304-a4b9-09187360c9e3)
to see that I am actually doing things correctly I wanted to first plot the configuration of the system but I didnt find a proper way to do so.

I imagine something like:
```python
grid = CylindricalSymGrid(radius=r_max, bounds_z=[0, z_max], shape=[100, 100])

# Define boundary conditions
bc_r = [
    {"derivative": 0},  # at r = 0 (symmetry condition)
    {"derivative": 0}   # at r = r_max (parallel field for r >> r_0)
]

bc_z = [
    {"value": f"{v_c} if r <= {r_0} * (1 - z / {h_cone}) else 0"},  # Cone boundary
    {"value": v_g}  # at z = z_max (top gate)
]

bcs = [bc_r, bc_z]

# printed the configuration (didnt find a way to do so)
grid.plot_with_bcs(bus)
```

Thank you very much!

**Comment by david-zwicker** [comment]
Boundary conditions are associated with fields or operators, not with grids. In any case, it looks as if you want to use a complex geometry, for which finite difference (the technique used by `py-pde`) is probably not suitable. You might want to look into finite element methods or related techniques.

**Reply by goldendijon** [reply]
the finite element packages that I saw was more complex and the support in cylindrical coordinates there wasn't trivial at all. that why I prefer to use py-pde, even if the precision wont be so high. do you have any idea how to implement it here?

thanks alot!
